source_roots.json

```diff
@@ -0,0 +1,123 @@
+[
```

Comment:
Verbose, but I think this is an improvement. It's far easier to reason with, and the verbosity is isolated to this single file and to the Pants admin when doing the initial setup.